### PR TITLE
add openTofu as a new deployer

### LINKED ISSUE #1247: add openTofu as a new deployer
Since OpenTofu 1.6 is now GA, we can now implement OpenTofu as a "deployer" type in addition to "terraform".

- [ ] Add new package `opentofu` under `deployment` maven module.
- [ ] Extend DeployerKind enum with OPEN_TOFU (opentofu)
- [ ] Implement functionality similar to "terraform-local".
- [ ] Move all common SystemCmd classes to `common` package.
- [ ] Test existing sample files using opentofu. 

Note - After this, we will also implement an application for opentofu similar to terraform-boot.

### PR DESCRIPTION
**Pull request by baixinsui**
fixes #1247

### CODE REVIEW

**Review by swaroopar — CHANGES_REQUESTED** [review]
Looks great. Just minor changes requested.

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/TerraformBootWebhookApi.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public class TerraformBootWebhookApi {
     public void deployCallback(
             @Parameter(name = "task_id", description = "task id")
             @PathVariable("task_id") String taskId,
-            @Valid @RequestBody TerraformResult result) {
+            @Valid @RequestBody TerraformExecutorResult result) {
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply modules/api/src/main/java/org/eclipse/xpanse/api/controllers/TerraformBootWebhookApi.java]: Restored.

File: modules/common/src/main/java/org/eclipse/xpanse/common/systemcmd/SystemCmdResult.java [hunk 2]
```diff
@@ -14,7 +14,7 @@
 public class SystemCmdResult {
 
     private String commandExecuted;
-    private boolean isCommandSuccessful;
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply modules/common/src/main/java/org/eclipse/xpanse/common/systemcmd/SystemCmdResult.java]: Restored.

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/modules/orchestrator/deployment/DeployerProvider.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+package org.eclipse.xpanse.modules.orchestrator.deployment;
+
+import org.eclipse.xpanse.modules.models.servicetemplate.enums.DeployerKind;
+
+/**
+ * This interface defines the Terraform provider supported by the plugin.
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply modules/orchestrator/src/main/java/org/eclipse/xpanse/modules/orchestrator/deployment/DeployerProvider.java]: Updated.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/terraformboot/model/TerraformResult.java [hunk 4]
```diff
@@ -3,7 +3,7 @@
  * RESTful Services to interact with Terraform-Boot runtime
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/terraformboot/model/TerraformResult.java]: Restored.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**